# ITMGT 25.03 EE
# Assignment 3A

This is a data-cleaning assignment and is the first of two parts. The second part (3B) will be given Monday or Tuesday next week. This notebook is uploaded to our Github repository along with a csv of fake, dirty data called **recweekdata.csv**.


The output of this assignment is to convert recweekdata.csv into processable data for statistics. Submit the following files through GitHub:
1. A cleaned recweekdata.csv called cleanrecweekdata.csv **(5 points)**
2. This notebook, containing the code (properly commented) that you used to clean the data **(5 points)**
3. An html file of summary statistics from each column called **stats.html**. The requirements for the summary statistics are described below. **(10 points)**

This entire assignment (3A) is worth 20 points.

Upload your output files to your Github repositories. Make sure to send an email to **joben.ilagan@gmail.com** once you're done.

This assignment is due on **Tuesday, July 9, 2019** at 12:00NN.

In [1]:
import csv

In [80]:
import csv #list of values that need not be cleaned (first name, last name and birth day)

first_name_list= []
last_name_list=[]
birth_day_list=[]
f= open('recweekdata.csv')
csv_f= csv.reader(f, delimiter= '	')

for row in csv_f:
    first_name_list.append(row[0])
    last_name_list.append(row[1])
    birth_day_list.append(row[3])
    
    

In [83]:
first_name_list.remove('firstname') #cleaning up headers

In [84]:
last_name_list.remove('lastname') #cleaning up headers

In [85]:
birth_day_list.remove('birthday') #cleaning up headers

In [87]:
birthday_tally={} #tallying birthday data

bd_set= set(birth_day_list)

for day in bd_set:
    birthday_tally[day]=0
    
for day in birth_day_list:
    birthday_tally[day] +=1
    
birthday_tally

{'5': 100,
 '2': 106,
 '11': 105,
 '16': 115,
 '8': 111,
 '27': 102,
 '22': 103,
 '9': 98,
 '12': 98,
 '23': 116,
 '13': 119,
 '14': 102,
 '26': 116,
 '1': 112,
 '17': 116,
 '24': 108,
 '6': 99,
 '20': 91,
 '15': 121,
 '10': 111,
 '21': 123,
 '18': 107,
 '4': 110,
 '3': 90,
 '7': 88,
 '19': 113,
 '25': 116,
 '28': 104}

In [3]:
import csv #imported birth month data

birth_month_list= []
f= open('recweekdata.csv')
csv_f= csv.reader(f, delimiter= '	')

for row in csv_f:
    birth_month_list.append(row[2])

In [4]:
birth_month_list #checking birth month data

['birthmonth',
 '11',
 'May',
 '5',
 'Dec',
 'Sep',
 'March',
 '11',
 'Aug',
 'July',
 '9',
 'December',
 '5',
 'January',
 '1',
 '6',
 '4',
 '1',
 'Apr',
 '12',
 '3',
 '7',
 '9',
 '4',
 'Oct',
 '5',
 '12',
 'March',
 '11',
 '10',
 'May',
 '4',
 'Jul',
 '12',
 'Oct',
 '7',
 'December',
 '2',
 '8',
 'May',
 '2',
 'Nov',
 'September',
 'Jan',
 'February',
 'January',
 'Dec',
 'August',
 'Apr',
 '10',
 'March',
 '8',
 '3',
 'Dec',
 '7',
 'May',
 '9',
 'May',
 'April',
 'April',
 'August',
 '9',
 'Dec',
 'Feb',
 'November',
 'Mar',
 'September',
 'Dec',
 'November',
 'November',
 'May',
 '8',
 'December',
 'Jun',
 'Oct',
 'Sep',
 '2',
 'May',
 '9',
 'December',
 '8',
 'Feb',
 '11',
 'Jun',
 'February',
 'Jan',
 'Nov',
 'August',
 'January',
 '3',
 'Mar',
 'Aug',
 'Dec',
 'Jun',
 'November',
 '9',
 'Mar',
 'Jan',
 'Jan',
 'May',
 '5',
 'April',
 'August',
 'Dec',
 'July',
 'Mar',
 '8',
 'April',
 'May',
 '8',
 'January',
 '12',
 'May',
 'May',
 'August',
 'March',
 'Jul',
 '1',
 'Feb',
 'De

In [5]:
unique_bm_list= set(birth_month_list) #isolating unique data

In [17]:
unique_bm_list #checking

{'1',
 '10',
 '11',
 '12',
 '2',
 '3',
 '4',
 '5',
 '6',
 '7',
 '8',
 '9',
 'Apr',
 'April',
 'Aug',
 'August',
 'Dec',
 'December',
 'Feb',
 'February',
 'Jan',
 'January',
 'Jul',
 'July',
 'Jun',
 'June',
 'Mar',
 'March',
 'May',
 'Nov',
 'November',
 'Oct',
 'Sep',
 'September',
 'birthmonth'}

In [6]:
list(unique_bm_list) #converting to list

['5',
 'Jul',
 'Dec',
 '11',
 '2',
 'August',
 'Sep',
 'Jan',
 '8',
 'Aug',
 '12',
 '9',
 'Mar',
 '1',
 'April',
 'November',
 'March',
 'Feb',
 '6',
 'January',
 'Apr',
 'June',
 '10',
 'Jun',
 'February',
 'birthmonth',
 'May',
 'September',
 'December',
 '4',
 'Nov',
 'Oct',
 '3',
 '7',
 'July']

In [7]:
unique_bm_dictionary = {name:None for name in unique_bm_list} #preparing dictionary data for input

In [20]:
unique_bm_dictionary #checking

{'Apr': None,
 'birthmonth': None,
 'Aug': None,
 'August': None,
 '11': None,
 'May': None,
 'September': None,
 '3': None,
 'December': None,
 '5': None,
 '9': None,
 'June': None,
 '6': None,
 'July': None,
 '10': None,
 'Jun': None,
 '2': None,
 '12': None,
 'February': None,
 'Sep': None,
 'Jul': None,
 'Mar': None,
 'Dec': None,
 'March': None,
 'Oct': None,
 '4': None,
 'April': None,
 'Feb': None,
 'January': None,
 'Jan': None,
 'November': None,
 '1': None,
 'Nov': None,
 '8': None,
 '7': None}

In [8]:
for i in unique_bm_dictionary: #cleaning data through input 
    final_name = input(i+": "+"Enter name: ")
    unique_bm_dictionary[i]=final_name

5: Enter name: May
Jul: Enter name: July
Dec: Enter name: December
11: Enter name: November
2: Enter name: February
August: Enter name: August
Sep: Enter name: September
Jan: Enter name: January
8: Enter name: August
Aug: Enter name: August
12: Enter name: December
9: Enter name: September
Mar: Enter name: March
1: Enter name: January
April: Enter name: April
November: Enter name: November
March: Enter name: March
Feb: Enter name: February
6: Enter name: June
January: Enter name: January
Apr: Enter name: April
June: Enter name: June
10: Enter name: October
Jun: Enter name: June
February: Enter name: February
birthmonth: Enter name: birthmonth
May: Enter name: May
September: Enter name: September
December: Enter name: December
4: Enter name: April
Nov: Enter name: November
Oct: Enter name: October
3: Enter name: March
7: Enter name: July
July: Enter name: July


In [9]:
import re #substituting data for input values

clean_bm_list = [re.sub(r''.join(month),r''.join(unique_bm_dictionary[month]),month) for month in birth_month_list]

In [75]:
clean_bm_list.remove('birthmonth') #removing heading

In [76]:
bm_tally= {} #tallying birth month data

bm_set= set(clean_bm_list)

for month in bm_set:
    bm_tally[month]=0
    
for month in clean_bm_list:
    bm_tally[month] +=1
    
bm_tally

{'October': 178,
 'May': 273,
 'September': 232,
 'December': 276,
 'August': 241,
 'January': 270,
 'June': 259,
 'April': 267,
 'November': 277,
 'February': 232,
 'July': 249,
 'March': 246}

In [11]:
import csv #importing csv data from birth year

birth_year_list= []
f= open('recweekdata.csv')
csv_f= csv.reader(f, delimiter= '	')

for row in csv_f:
    birth_year_list.append(row[4])

In [25]:
birth_year_list #Checking

['birthyear',
 '1998',
 '2001',
 '1997',
 '97',
 '97',
 '1999',
 '98',
 '99',
 '97',
 '2001',
 '98',
 '97',
 '1999',
 '1999',
 '98',
 '99',
 '97',
 '1997',
 '1997',
 '97',
 '99',
 '1999',
 '97',
 '97',
 '97',
 '1999',
 '1997',
 '2000',
 '1999',
 '1999',
 '99',
 '99',
 '99',
 '1999',
 '2001',
 '1998',
 '2000',
 '2001',
 '1997',
 '1998',
 '99',
 '98',
 '1999',
 '98',
 '97',
 '98',
 '1999',
 '99',
 '97',
 '2001',
 '98',
 '1999',
 '2000',
 '98',
 '1998',
 '1998',
 '2001',
 '97',
 '98',
 '99',
 '98',
 '2000',
 '2000',
 '97',
 '97',
 '1997',
 '1997',
 '2001',
 '1998',
 '98',
 '99',
 '1999',
 '1998',
 '1997',
 '1998',
 '1997',
 '2000',
 '2000',
 '2000',
 '99',
 '97',
 '1997',
 '2000',
 '1999',
 '1999',
 '98',
 '1998',
 '97',
 '1998',
 '99',
 '97',
 '97',
 '1998',
 '97',
 '2001',
 '1999',
 '1997',
 '98',
 '98',
 '99',
 '97',
 '2001',
 '2000',
 '2001',
 '99',
 '97',
 '2001',
 '99',
 '1998',
 '99',
 '2001',
 '97',
 '97',
 '99',
 '98',
 '97',
 '1997',
 '1999',
 '1997',
 '97',
 '2000',
 '2001',
 '

In [12]:
by_dict= set(birth_year_list) #isolating unique values


In [13]:
unique_by_list= list(by_dict) #converting unique values to list

In [14]:
unique_by_dictionary = {name:None for name in unique_by_list} #preparing for data cleaning

In [15]:
for i in unique_by_dictionary: #inputting cleaned values
    final_name = input(i+": "+"Enter name: ")
    unique_by_dictionary[i]=final_name

97: Enter name: 1997
1998: Enter name: 1998
birthyear: Enter name: birthyear
1999: Enter name: 1999
2000: Enter name: 2000
98: Enter name: 1998
99: Enter name: 1999
1997: Enter name: 1997
2001: Enter name: 2001


In [16]:
import re #substituting data with clean inputs

clean_by_list = [re.sub(r''.join(year),r''.join(unique_by_dictionary[year]),year) for year in birth_year_list]

In [73]:
clean_by_list.remove('birthyear') #removing heading

In [74]:
by_tally= {} #tallying year data

by_set= set(clean_by_list)

for year in by_set:
    by_tally[year]=0
    
for year in clean_by_list:
    by_tally[year] +=1
    
by_tally

{'1999': 756, '1998': 746, '2000': 380, '1997': 756, '2001': 362}

In [18]:
import csv #importing course data

course_list= []
f= open('recweekdata.csv')
csv_f= csv.reader(f, delimiter= '	')

for row in csv_f:
    course_list.append(row[5])

In [19]:
un_courses= set(course_list) #isolating unique course values

In [20]:
un_course_list= list(un_courses) #converting course data to list

In [21]:
un_course_dict = {name:None for name in un_course_list} #preparing for data cleaning

In [22]:
for i in un_course_dict: #inputting correct value names
    final_name = input(i+": "+"Enter name: ")
    un_course_dict[i]=final_name

BS CTM: Enter name: BS Communications Technology Management
ITE: Enter name: BS Information Technology Entrepreneurship
Information Technology Entrepreneurship: Enter name: BS Information Technology Entrepreneurship
BS LM: Enter name: BS Legal Management
BS ME: Enter name: BS Management Engineering
MGT: Enter name: BS Management
Communications Technology: Enter name: BS Communications Technology Management
course: Enter name: course
Legal Management: Enter name: BS Legal Management
CTM: Enter name: BS Communications Technology Management
BS MGT: Enter name: BS Management
ME: Enter name: BS Management Engineering
Management Engineering: Enter name: BS Management Engineering
BS ITE: Enter name: BS Information Technology Entrepreneurship
LM: Enter name: BS Legal Management
Management: Enter name: BS Management


In [23]:
import re #substituting course values to cleaned inputs

clean_course_list = [re.sub(r''.join(course),r''.join(un_course_dict[course]),course) for course in course_list]

In [71]:
clean_course_list.remove('course') #removing heading

In [72]:
course_tally= {} #tallying course data

course_set= set(clean_course_list)

for course in course_set:
    course_tally[course]=0
    
for course in clean_course_list:
    course_tally[course] +=1
    
course_tally

{'BS Legal Management': 585,
 'BS Management Engineering': 607,
 'BS Communications Technology Management': 589,
 'BS Management': 629,
 'BS Information Technology Entrepreneurship': 590}

In [25]:
import csv #importing cellphone data
 
cpno_list= []
f= open('recweekdata.csv')
csv_f= csv.reader(f, delimiter= '	')


for row in csv_f:
    cpno_list.append(row[6])


    

In [26]:
un_cpno_list=list(set(cpno_list)) #converting to list unique values


In [27]:
un_cpno_list #checking

['+63 917 929 4398',
 '+63 918 531 2515',
 '927 485 1284',
 '+63 917 857 7520',
 '+63 917 731 3274',
 '+63 927 329 9292',
 '+63 918 111 2365',
 '+63 927 725 9531',
 '+63 918 914 3596',
 '927 105 7640',
 '927 596 1752',
 '+63 927 942 1394',
 '+63 918 305 5557',
 '917 793 4237',
 '918 645 9539',
 '917 574 1808',
 '+63 918 317 1448',
 '927 647 2917',
 '+63 918 989 9777',
 '+63 917 333 9074',
 '927 731 3323',
 '+63 917 883 2016',
 '918 968 3488',
 '+63 927 563 4960',
 '+63 918 704 3039',
 '917 928 8674',
 '918 511 4076',
 '917 557 5935',
 '917 166 8658',
 '+63 917 385 3998',
 '917 653 7150',
 '+63 917 168 1549',
 '+63 918 695 2117',
 '927 488 3954',
 '917 242 3611',
 '918 597 2109',
 '+63 927 603 5096',
 '+63 918 555 9070',
 '918 153 2402',
 '918 592 7760',
 '918 821 7480',
 '+63 917 707 6263',
 '+63 927 146 1580',
 '917 135 6232',
 '+63 917 199 2397',
 '918 339 1304',
 '927 830 1201',
 '917 575 1571',
 '+63 917 103 4403',
 '+63 927 479 1685',
 '917 693 4040',
 '927 196 8999',
 '+63 917 54

In [28]:
import re #replacing values to uniform format
uniform_no_list=[]

for num in un_cpno_list:
    replaced_num= re.sub(r"^9([0-9][0-9])", r"+63 9\1", num)
    uniform_no_list.append(replaced_num)
    
    

In [75]:
uniform_no_list #checking

['+63 927 792 8470',
 '+63 917 634 5100',
 '+63 918 647 2458',
 '+63 918 117 8158',
 '+63 917 586 5535',
 '+63 918 629 5268',
 '+63 917 379 9404',
 '+63 918 373 3234',
 '+63 918 848 1893',
 '+63 917 761 3830',
 '+63 918 191 7948',
 '+63 918 240 8941',
 '+63 917 166 3062',
 '+63 918 883 4422',
 '+63 918 510 5817',
 '+63 918 455 4185',
 '+63 927 699 3694',
 '+63 917 627 4750',
 '+63 917 764 8229',
 '+63 918 469 3848',
 '+63 918 399 7923',
 '+63 918 200 4621',
 '+63 927 621 3475',
 '+63 917 448 4860',
 '+63 917 668 9753',
 '+63 917 560 3594',
 '+63 918 340 9046',
 '+63 917 506 7963',
 '+63 917 526 1163',
 '+63 917 954 6667',
 '+63 917 840 6344',
 '+63 918 162 2673',
 '+63 927 942 1394',
 '+63 918 561 4308',
 '+63 918 595 9543',
 '+63 927 545 6383',
 '+63 927 125 3542',
 '+63 927 685 6676',
 '+63 917 464 7561',
 '+63 927 313 4542',
 '+63 917 811 6162',
 '+63 918 424 3370',
 '+63 927 138 7580',
 '+63 927 867 3896',
 '+63 917 313 8195',
 '+63 917 864 9783',
 '+63 927 211 1491',
 '+63 918 713

In [29]:
network_tally= {"Smart":0, "Globe":0} #tallying network data

    
for num in uniform_no_list:
    globe= re.search(r'[+]63\s9[1-2]7\s[0-9]{3}\s[0-9]{4}', num)
    smart= re.search(r'[+]63\s918\s[0-9]{3}\s[0-9]{4}', num)
    if smart:
        network_tally["Smart"]+=1
    if globe:
        network_tally["Globe"]+=1
        

In [44]:
network_tally #checking

{'Smart': 955, 'Globe': 2045}

In [42]:
clean_by_list[1] #trying to isolate value

'1998'

In [86]:
import csv #combining into new csv file

row_format= zip(first_name_list, last_name_list, clean_bm_list, birth_day_list, clean_by_list, clean_course_list, uniform_no_list)
with open('recweekdata.csv', 'r') as csv_file:
    csv_reader = csv.reader(csv_file, delimiter= '	')
    

    with open('newrecweekdata.csv', 'w') as new_file:
        csv_writer = csv.writer(new_file, quoting=csv.QUOTE_MINIMAL)
        csv_writer.writerow(("firstname", "lastname", "birthmonth", "birthday", "birthyear", "course","cellphonenumber"))
        for row in row_format:
            csv_writer.writerow(row)
            
                               



 

In [55]:
#mistake in trying to get format without using zip

[[['firstname',
   'name1',
   'name2',
   'name3',
   'name4',
   'name5',
   'name6',
   'name7',
   'name8',
   'name9',
   'name10',
   'name11',
   'name12',
   'name13',
   'name14',
   'name15',
   'name16',
   'name17',
   'name18',
   'name19',
   'name20',
   'name21',
   'name22',
   'name23',
   'name24',
   'name25',
   'name26',
   'name27',
   'name28',
   'name29',
   'name30',
   'name31',
   'name32',
   'name33',
   'name34',
   'name35',
   'name36',
   'name37',
   'name38',
   'name39',
   'name40',
   'name41',
   'name42',
   'name43',
   'name44',
   'name45',
   'name46',
   'name47',
   'name48',
   'name49',
   'name50',
   'name51',
   'name52',
   'name53',
   'name54',
   'name55',
   'name56',
   'name57',
   'name58',
   'name59',
   'name60',
   'name61',
   'name62',
   'name63',
   'name64',
   'name65',
   'name66',
   'name67',
   'name68',
   'name69',
   'name70',
   'name71',
   'name72',
   'name73',
   'name74',
   'name75',
   'name76',
   'n

In [95]:
f = open('stats.html','w') #html


stats= """<html>
    <body>
         <h2>Summary Statistics</h2>
         <table>
             <tr><th>Stat</th><th>Value</th></tr>
             <tr><td>People born in each month</td><td>{'October': 178,
                                                         'May': 273,
                                                         'September': 232,
                                                         'December': 276,
                                                         'August': 241,
                                                         'January': 270,
                                                         'June': 259,
                                                         'April': 267,
                                                         'November': 277,
                                                         'February': 232,
                                                         'July': 249,
                                                         'March': 246}</td></tr>
             <tr><td>People born on each day</td><td>{'5': 100,
                                                     '2': 106,
                                                     '11': 105,
                                                     '16': 115,
                                                     '8': 111,
                                                     '27': 102,
                                                     '22': 103,
                                                     '9': 98,
                                                     '12': 98,
                                                     '23': 116,
                                                     '13': 119,
                                                     '14': 102,
                                                     '26': 116,
                                                     '1': 112,
                                                     '17': 116,
                                                     '24': 108,
                                                     '6': 99,
                                                     '20': 91,
                                                     '15': 121,
                                                     '10': 111,
                                                     '21': 123,
                                                     '18': 107,
                                                     '4': 110,
                                                     '3': 90,
                                                     '7': 88,
                                                     '19': 113,
                                                     '25': 116,
                                                     '28': 104}</td></tr>
             <tr><td>People born in each year</td><td>{'1999': 756, '1998': 746, '2000': 380, '1997': 756, '2001': 362}</td></tr>
             <tr><td>People coming from each course</td><td>{'BS Legal Management': 585,
                                                             'BS Management Engineering': 607,
                                                             'BS Communications Technology Management': 589,
                                                             'BS Management': 629,
                                                             'BS Information Technology Entrepreneurship': 590}</td></tr>
             <tr><td>People having numbers from Globe and Smart</td><td>{'Smart': 955, 'Globe': 2045}</td></tr>
         </table>
    </body>
</html>
"""
f.write(stats)
f.close()

# Required Summary Statistics
1. How many people were born in each month?
2. How many people were born on each day?
3. How many people were born in each year?
4. How many people come from each course?
5. How many people have numbers from Globe (0917 or 0927) and Smart (0918)? *Note: it's up to you how to format this output*

Answer these questions in an html file called stats.html. The format of the html is as follows (in template form, so you can use this as format string in your program, though feel free to modify with variable names if you find it easier to do so):


```
<html>
    <body>
         <h2>Summary Statistics</h2>
         <table>
             <tr><th>Stat</th><th>Value</th></tr>
             <tr><td>People born in each month</td><td>{}</td></tr>
             <tr><td>People born on each day</td><td>{}</td></tr>
             <tr><td>People born in each year</td><td>{}</td></tr>
             <tr><td>People coming from each course</td><td>{}</td></tr>
             <tr><td>People having numbers from Globe and Smart</td><td>{}</td></tr>
         </table>
    </body>
</html>
```